### PR TITLE
docs(mongodb): document new functionalities

### PR DESCRIPTION
**Pull request by kamilmysliwiec**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: content/techniques/mongo.md [hunk 1]
```diff
@@ -173,6 +171,94 @@ export class CatsService {
...
+          return schema;
+        },
+      },
+    ]),
+  ],
+})
+export class AppModule {}
+```
+
+To register a plugin for all schemas at once, call the `.plugin()` method of the `Connection` object. In order to access the connection before models are created, use the `connectionFactory`:
```
Comment (johnbiundo) [inline content/techniques/mongo.md]: Suggestion:

Replace: In order to access the connection before models are created, use the `connectionFactory`:

with: You should access the connection before models are created; to do this, use the `connectionFactory`:

File: content/techniques/mongo.md [hunk 2]
```diff
@@ -173,6 +171,94 @@ export class CatsService {
...
+      }
+    ]),
+  ],
+})
+export class AppModule {}
+```
+
+#### Plugins
+
+To register a [plugin](https://mongoosejs.com/docs/plugins.html) for a given schema, use the `forFeatureAsync()` method.
```
Comment (johnbiundo) [inline content/techniques/mongo.md]: Suggestion: add the following:

As with `pre()` and `post()` hooks, use the `useFactory` technique to access a schema in order to invoke the `plugin()` method.

File: content/techniques/mongo.md [hunk 3]
```diff
@@ -173,6 +171,94 @@ export class CatsService {
 }
 ```
 
+#### Hooks (middleware)
+
+Middleware (also called pre and post hooks) are functions which are passed control during execution of asynchronous functions. Middleware is specified on the schema level and is useful for writing plugins ([source](https://mongoosejs.com/docs/middleware.html)). Calling `pre()` or `post()` after compiling a model does not work in Mongoose. In order to register a hook **before** model registration, use the `forFeatureAsync()` method.
```
Comment (johnbiundo) [inline content/techniques/mongo.md]: Suggestion: to be a little more explicit, please consider this change:
To register a hook **before** model registration, use the `forFeatureAsync()` method of the `MongooseModule` along with a factory provider (i.e., `useFactory`).  With this technique, you can access a `schema` object, then use the `pre()` or `post()` method to register a hook on that schema.  See example below:

**Comment by kamilmysliwiec** [comment]
Look good @johnbiundo. Thank you!!

**Pull request merged**